### PR TITLE
Fixed sessionToken installer analytics.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2228" title="DX-2228" target="_blank">BugDX-2228</a>  Investigate missing session ID in state tool analytics
  </td></table>
  <br />

### CODE REVIEW

**Comment by mitchell-as** [comment]
Test failures are not due to this PR. They are due to existing, known issues and the usual timeouts.

File: cmd/state-installer/cmd.go [hunk 1]
```diff
@@ -123,6 +123,19 @@ func main() {
...
 
+	// Store sessionToken to config
+	for _, envVar := range []string{constants.OverrideSessionTokenEnvVarName, constants.SessionTokenEnvVarName} {
+		sessionToken, ok := os.LookupEnv(envVar)
+		if !ok {
+			continue
+		}
+		err := cfg.Set(anaConst.CfgSessionToken, sessionToken)
+		if err != nil {
+			logging.Error("Unable to set session token: " + errs.JoinMessage(err))
```
Comment (MDrakos) [inline cmd/state-installer/cmd.go:134]: We _may_ want to report this to rollbar so we can more easily diagnose if the issue of a missing session token comes up again.

File: test/integration/install_scripts_int_test.go [hunk 2]
```diff
@@ -260,6 +264,21 @@ func (suite *InstallScriptsIntegrationTestSuite) assertCorrectVersion(ts *e2e.Se
...
 }
 
+func (suite *InstallScriptsIntegrationTestSuite) assertAnalytics(ts *e2e.Session) {
+	// Verify analytics reported the correct sessionToken.
+	sessionTokenFound := false
+	events := parseAnalyticsEvents(suite, ts)
+	suite.Require().NotEmpty(events)
+	for _, event := range events {
+		if event.Category == anaConst.CatInstallerFunnel && event.Dimensions != nil {
+			suite.Assert().NotEmpty(*event.Dimensions.SessionToken)
```
Comment (MDrakos) [inline test/integration/install_scripts_int_test.go:274]: nit: We _could_ assert the equality of the token value, not just is presence, as the comment above suggests.

**Review by MDrakos — APPROVED** [review]
Looks good to me, just a couple of nits.

File: cmd/state-installer/cmd.go (hunk 1 above)
Comment (mitchell-as) [reply cmd/state-installer/cmd.go]: Yes, that's a good idea.

File: test/integration/install_scripts_int_test.go (hunk 2 above)
Comment (mitchell-as) [reply test/integration/install_scripts_int_test.go:274]: I've updated the comment. I am guilty of copy-pasting the code from another test :/ The important thing is that it's not empty.

**Review by MDrakos — APPROVED** [review]
